### ISSUE TITLE
Compose on windows does not support Utf-8 BOM on 1.21.2

### ISSUE THREAD

**Issue by thomasbruggink** [docker/compose#6015]
## Description of the issue
When the encoding of a compose file is UTF-8 BOM docker compose up results in the following error:
`ERROR: In file '.\docker-compose.yml', service 'ï»¿version' must be a mapping not a string.`
When using version 1.15.0-rc1 the issue does not occur.

```
Output of "docker-compose version"
```
Working version: docker-compose version 1.15.0-rc1, build 2188098f
Failing verson: docker-compose version 1.21.2, build a1334711
```
Output of "docker version"
```
Docker version 17.05.0-ce, build 89658be
```
Output of "docker-compose config"
```
Stored with encoding UTF8 BOM

version: "3"
services:
  consul:
    image: consul:latest
    networks:
      - nat
networks:
  nat:
    external: true

## Steps to reproduce the issue

1. Save the content above in UTF8 BOM
2. docker-compose up

### Observed result
ERROR: In file '.\docker-compose.yml', service 'ï»¿version' must be a mapping not a string

### Expected result
Creating consul_1    ... done

## Additional information
Windows Server 2016. Compose was installed by extracting the executable. I am not using docker toolbox.
This issue does not occur on the Linux version of compose

**Comment by shin-** [comment]
This is weird because we definitely [test for it](https://github.com/docker/compose/blob/master/tests/unit/config/config_test.py#L1681-L1694), on Windows as well (through AppVeyor).

@thomasbruggink Does #6017 help? Do you see the same issue using 1.20.1?

**Comment by thomasbruggink** [comment]
1.20.x gives me the same issue. The issue perhaps doesn't seem the be in docker-compose itself. When I tried to run docker-compose under the debugger it didn't matter if I used 1.15/1.20/1.21 all gave the same error.
I also noticed that I don't actually hit the catch block your fix suggested but just return without an issue. The encoding given is the following:
![image](https://user-images.githubusercontent.com/16775318/41320704-23ad1e6c-6ea1-11e8-976b-2724ae404062.png)
